packages/venia-ui/lib/components/ForgotPassword/forgotPassword.js

```diff
@@ -1,61 +1,64 @@
-import React, { Component, Fragment } from 'react';
-import PropTypes from 'prop-types';
+import React, { Fragment, useCallback } from 'react';
+import { bool, func, shape, string } from 'prop-types';
 
-import classify from '../../classify';
+import { mergeClasses } from '../../classify';
 import ForgotPasswordForm from './ForgotPasswordForm';
 import FormSubmissionSuccessful from './FormSubmissionSuccessful';
 import defaultClasses from './forgotPassword.css';
 
-class ForgotPassword extends Component {
-    static propTypes = {
-        classes: PropTypes.shape({
-            instructions: PropTypes.string
-        }),
-        completePasswordReset: PropTypes.func.isRequired,
-        email: PropTypes.string,
-        initialValues: PropTypes.shape({
-            email: PropTypes.string
-        }),
-        isInProgress: PropTypes.bool,
-        onClose: PropTypes.func.isRequired,
-        resetPassword: PropTypes.func.isRequired
-    };
-
-    handleFormSubmit = async ({ email }) => {
-        this.props.resetPassword({ email });
-    };
-
-    handleContinue = () => {
-        const { completePasswordReset, email, onClose } = this.props;
+const INSTRUCTIONS = 'Enter your email below to receive a password reset link.';
 
+const ForgotPassword = props => {
+    const {
+        completePasswordReset,
+        email,
+        initialValues,
+        isInProgress,
```

Comment:
I think there is room to improve this workflow. Right now we set `isInProgress` to true when you submit the form and then set it to false when you click the "Continue Shopping" button. I think this UX is strange. We should probably close the menu entirely (since that's what "Continue Shopping" would do, I imagine) or we should change the button to say `Back` and then present the user with the navigation menu, sign in page, or something else. I think we could probably pass one of the `show` callbacks from navigation.js down to the forgotPassword form and use that when the user submits/clicks back.

[![Image from Gyazo](https://i.gyazo.com/edaa580ebd26cdf1a5cb68955d79cf62.gif)](https://gyazo.com/edaa580ebd26cdf1a5cb68955d79cf62)

Reply:
Good catch. You're right, `Continue Shopping` should reset the view state (back to `MENU`) and close the drawer. The wording of the button is design's preference, so I don't think I'll change it.

Reply:
This has been addressed.